pkgs/development/tools/sumneko-lua-language-server/default.nix

```
@@ -0,0 +1,47 @@
+{ stdenv, fetchFromGitHub, ninja, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "sumneko-lua-language-server";
+  version = "1.9.0";
+
+  src = fetchFromGitHub {
+    owner = "sumneko";
+    repo = "lua-language-server";
+    rev = "${version}";
+    sha256 = "0d93ikxs0c5fm9k57xbzgphm90dnfqq3nz2bc3bqxi7gzyldb9z6";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    ninja
+    makeWrapper
+  ];
+
+  patches = [
+    ./sumneko_home_expansion.patch
+  ];
+
+  buildPhase = ''
+    cd 3rd/luamake
+    ninja -f ninja/linux.ninja
+    cd ../..
+    ./3rd/luamake/luamake rebuild
+  '';
```

I would like to use the ninja setup-hook. Can we do something like to do that:

```
```suggestion
  preBuildPhase = ''
    cd 3rd/luamake
  '';
  
  ninjaFlags = "-f ninja/linux.ninja";

  postBuildPhase = ''
    cd ../..
    ./3rd/luamake/luamake rebuild
  '';
```

---

I tried that before and I'm not sure why it isn't working
```
build flags: -j8 -l8 -f ninja/linux.ninja
ninja: error: loading 'ninja/linux.ninja': No such file or directory
ninja: error: loading 'ninja/linux.ninja': No such file or directory
builder for '/nix/store/i8xzv17v3n4z28yr6786k5a2gjvl78xs-sumneko-lua-language-server-1.9.0.drv' failed with exit code 1
```

---

Did you add the cd in preBuild?

---

This is exactly what I did: https://github.com/mjlbach/nixpkgs/blob/sumneko_build_failure/pkgs/development/tools/sumneko-lua-language-server/default.nix

```
https://github.com/mjlbach/nixpkgs/blob/78628a94ca35b49b06d938dbe010c399ef765548/pkgs/development/tools/sumneko-lua-language-server/default.nix#L24-L34
  patching sources
  applying patch /nix/store/3fsc6p3wsv5ainwhjm22agzx13aqx99z-sumneko_home_expansion.patch
  patching file main.lua
  configuring
  no configure script, doing nothing
  building
  build flags: -j8 -l8 -f ninja/linux.ninja
  ninja: error: loading 'ninja/linux.ninja': No such file or directory
error: --- Error ---------------------------------------------------------------------------------------------------------------------------------------------------------------- nix
build of '/nix/store/i8xzv17v3n4z28yr6786k5a2gjvl78xs-sumneko-lua-language-server-1.9.0.drv' failed
```

---

Because it is preBuild and postBuild without phase. Also if you execute ninja you need to add ``-j$NIX_BUILD_CORES -l$NIX_BUILD_CORES``

---

When I add those to ninja flags I get an error, am I doing it wrong?
```nix
  ninjaFlags = [
    "-f ninja/linux.ninja"
    "-j$NIX_BUILD_CORES"
    "-l$NIX_BUILD_CORES"
    ];
```

---

Is this not already done? https://github.com/mjlbach/nixpkgs/blob/a57ba9ea13c2f09a467e9237c516ba22a09ae93f/pkgs/development/tools/build-managers/ninja/setup-hook.sh#L6-L13